### PR TITLE
add ddns-scripts for huaweicloud

### PR DESCRIPTION
**Pull request by sxlehua**
Maintainer:  @feckert
Compile tested: all supported targets
Run tested: mediatek/filogic

Description:
add ddns-scripts for huaweicloud

### CODE REVIEW

File: net/ddns-scripts/Makefile [hunk 1]
```diff
@@ -386,6 +395,7 @@ define Package/ddns-scripts-services/install
 	rm $(1)/usr/share/ddns/default/ns1.com.json
 	rm $(1)/usr/share/ddns/default/one.com.json
 	rm $(1)/usr/share/ddns/default/porkbun.com-v3.json
+	rm $(1)/usr/share/ddns/default/huaweicloud.com.json.json
```
Comment (feckert) [inline net/ddns-scripts/Makefile:398]: json.json?

File: net/ddns-scripts/files/usr/share/ddns/default/huaweicloud.com.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+	"name": "huaweicloud.com",
+	"ipv4": {
+		"url": "update_huaweicloud_com.sh"
+	},
+	"ipv6": {
+		"url": "update_huaweicloud_com.sh"
+	}
+}
```
Comment (feckert) [inline net/ddns-scripts/files/usr/share/ddns/default/huaweicloud.com.json:9]: newline is missing

File: net/ddns-scripts/files/usr/lib/ddns/update_huaweicloud_com.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
+#!/bin/sh
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_huaweicloud_com.sh:1]: Please use <tabs> for indentation.

File: net/ddns-scripts/files/usr/lib/ddns/update_huaweicloud_com.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+# Check inputs
+[ -z "$username" ] && write_log 14 "Configuration error! [username] cannot be empty"
+[ -z "$password" ] && write_log 14 "Configuration error! [password] cannot be empty"
+
+[ -z "$CURL" ] && [ -z "$CURL_SSL" ] && write_log 14 "huaweicloud API require cURL with SSL support. Please install"
+command -v openssl >/dev/null 2>&1 || write_log 14 "huaweicloud API require openssl-util support. Please install"
+
+# public variable
+local __HOST __DOMAIN __TYPE __ZONE_ID __RECORD_ID
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_huaweicloud_com.sh:25]: `local` is not allowed for ash in global context as I know?

**Review by feckert — CHANGES_REQUESTED** [review]
Also please bump the PKG_RELEASE by one,

File: net/ddns-scripts/Makefile (hunk 1 above)
Comment (sxlehua) [reply net/ddns-scripts/Makefile:398]: I will fix it

**Comment by 1715173329** [comment]
Please format your commit title and SoB according to https://openwrt.org/submitting-patches.

**Comment by sxlehua** [comment]
> Please format your commit title and SoB according to https://openwrt.org/submitting-patches.

is ok?

**Comment by feckert** [comment]
Thanks merged!

**Pull request merged**